### PR TITLE
Update API page

### LINKED ISSUE #206: automation: ensure all tutorials runs on latest fluvio release
All tutorial on master branch should run with dev release of fluvio.
All tutorial on stable branch should run with latest release of fluvio.

This should be part of website CI.

### LINKED ISSUE #325: Move other client tutorials from InfinyOn to Fluvio
It's a little confusing that the language APIs have no tutorial on fluvio.io.

They are [hosted on the InfinyOn site](https://www.infinyon.com/tutorials/rust/), but I don't believe they belong there.

### PR DESCRIPTION
**Pull request by tjtelan**
Resolves #325 
Resolves #206 

* Remove link to tutorial in Infinyon.
* New navigation
  * Restructure API page to emphasize a separation of instructions for installation, and snippets for using the language clients
  * Expose individual docs pages at the top level.
* New examples with a much more concise explanation. Fewer technical details, and more emphasis on helping users write copy/paste code.
* Add CI for API examples
  * Using Docker to validate installation instructions
  * Using Docker Compose for mounting config files, to validate example executes without error

No change to Overview page in this PR

image

### CODE REVIEW

**Comment by simlay** [comment]
Might as well also add dependabot.

File: code/rust/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,6 @@
+FROM rust:1.61
```
[the remark this answers is not in the document]
Comment (tjtelan) [reply code/rust/Dockerfile:1]: To be honest, it was just convenient for me to repeatedly test installation locally with docker-compose. And I knew the github runners had docker-compose installed.

**Review by ajhunyady — APPROVED** [review]
Looks good.

File: code/rust/Dockerfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (tjtelan) [reply code/rust/Dockerfile:2]: I added a note to all the Dockerfile and docker-compose to make it clear that they are being used by CI

**Pull request merged**